travel-planner/travel-planner-backend/routes/maps.js

```diff
@@ -5,9 +5,62 @@ const bodyParser = require("body-parser");
 router.use(bodyParser.json());
 const Parse = require("parse/node");
 const axios = require("axios");
+// const {Client} = require("@googlemaps/google-maps-services-js");
 
 // create delete pin endpoint?
 
+router.post("/getDirections", async (req, res) => {
+	const type = req.body.type;
+	const origin = req.body.origin;
+	const destination = req.body.destination;
+	const distance = req.body.distance;
+	const duration = req.body.duration;
+	const directionsArr = req.body.directions;
+
+	// Save in the database
+	const Directions = Parse.Object.extend("Directions");
+	const Maps = Parse.Object.extend("Maps");
+
+	try {
+		const direction = new Directions();
+		const mapPointer = new Maps().set("objectId", req.body.mapId);
+
+		direction.set("Origin", origin);
+		direction.set("Destination", destination);
+		direction.set("TravelMode", type);
+		direction.set("Duration", duration);
+		direction.set("Distance", distance);
+		direction.set("Directions", directionsArr);
+		direction.set("Map", mapPointer);
+
+		direction.save().then(() => {
+			res.status(200).send({ message: "Success!" });
+		});
+	} catch (error) {
+		console.log("Error: ", error);
+		res.status(400).send({ message: error, type: "Error" });
+	}
+});
+
+router.post("/getMapDirections", async (req, res) => {
```

Comment:
What does "getMapDirections" mean? Kinda confused, and we have "getDirections" api as well

Reply:
Is it to get the map for the current direction plan? Could we rename it to have better clarity?

Reply:
This is to get all the previous directions saved for that map. I've renamed it "getAllSavedDirections"

Reply:
looks like this one here is the real getter, and the other is more of a "createMapDirections"